script.js

```diff
@@ -1,4 +1,283 @@
+var counter = 0;
+while (counter < deck.length) {
+  if (
+    deck[counter].name == "jack" ||
+    deck[counter].name == "queen" ||
+    deck[counter].name == "king"
+  ) {
+    deck[counter].rank = 10;
+  }
+
+  counter = counter + 1;
+}
+
+// shuffle deck
+var shuffledDeck = shuffleCards(deck);
+
+// game modes  input name , player hand  ,
+var gameMode = "inputName";
+var playerName = "";
+var yourBetAmount = 0;
+var playerScore = 0;
+var card1 = {};
+var card2 = {};
+var dealerCards = [];
+
+var printCard = function (card) {
+  cardPic = `<div id="blackcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+
+  if (card.suit == "hearts" || card.suit == "diamonds") {
+    cardPic = `<div id="redcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+  }
+
+  return cardPic;
+};
+
+// modes inputName , betAmount, deal2cards , decidehit/Stay , dealerMode 2card autohit
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "inputName") {
+    playerName = input;
+    gameMode = "betAmount";
+    return (
+      "🥂 Hi " +
+      playerName +
+      ", welcome to the 'No Limit Blackjack Table'" +
+      "<br>" +
+      "👉 Please type in your betting amount 💰💰💰 in the orange bar above !"
+    );
+  }
+  if (gameMode == "betAmount") {
+    yourBetAmount = input;
+    gameMode = "ace1/11";
+    return (
+      "🥂 Great " +
+      playerName +
+      ", today seems to be your lucky day, 💰 you have bet: " +
+      yourBetAmount +
+      "<br>" +
+      "🤞 You can decide the Rank of Ace to be either 1 or 11" +
+      "<br>" +
+      "👉 Type in your choice: 1 or 11 in the orange bar above."
+    );
+  }
+
+  if (gameMode == "ace1/11") {
+    gameMode = "dealCards";
+    var counter = 0;
+    while (counter < shuffledDeck.length) {
+      if (shuffledDeck[counter].name == "ace") {
+        shuffledDeck[counter].rank == input;
+      }
+      counter = counter + 1;
+    }
+    return (
+      "🃎 Ace will have Rank: " +
+      input +
+      " for this round" +
+      "<br>" +
+      "♣️♠️♦️❤️ So it's now the Blackjack time. 👍 Press the 'Submit' button now" +
+      "<br>" +
+      "🃏🃏 Dealer will shuffle the deck and pull out two cards for you."
+    );
+  }
+
+  if (gameMode == "dealCards") {
+    card1 = shuffledDeck.pop();
+    card2 = shuffledDeck.pop();
+    playerScore = card1.rank + card2.rank;
+    if (playerScore == 21) {
+      gameMode = "endGame/newGame";
+      decision1 =
+        "perfect Blackjack , congrats" + "<br>" + "select end/continue ?";
+    }
+    if (playerScore > 21) {
+      gameMode = "endGame/newGame";
+      decision1 = "Bust, you lost your bet" + "<br>" + "select end/continue ?";
+    }
+    if (playerScore < 21) {
+      gameMode = "hitStay";
+      decision1 =
+        ". you want 'hit' or 'stay' ?" +
+        "<br>" +
+        "Type in your decision 'hit' or 'stay' in the orange bar above.";
+    }
```

Comment:
Would you prefer to use an `if`-`else if`-`else` chain for this? Consider the following:
```
if (playerScore == 21) {
  gameMode = "endGame/newGame";
  decision1 =
    "perfect Blackjack , congrats" + "<br>" + "select end/continue ?";
}
else if (playerScore > 21) {
  gameMode = "endGame/newGame";
  decision1 = "Bust, you lost your bet" + "<br>" + "select end/continue ?";
}
else {
  // we use else here because the previous 2 conditions are considered:
  // (1) equal 21, (2) greater than 21
  // all other cases, playerScore should be less than 21 so an else will do
  // instead of manually writing (playerScore < 21)
  gameMode = "hitStay";
  decision1 =
    ". you want 'hit' or 'stay' ?" +
    "<br>" +
    "Type in your decision 'hit' or 'stay' in the orange bar above.";
}
```